src/LanguageServer/Protocol/Protocol/Internal/Converters/ImageElementConverter.cs

```diff
@@ -30,7 +33,11 @@ public override ImageElement Read(ref Utf8JsonReader reader, Type typeToConvert,
 
                 if (reader.TokenType == JsonTokenType.PropertyName)
                 {
-                    var propertyName = reader.GetString();
+                    var valueLength = reader.HasValueSequence ? reader.ValueSequence.Length : reader.ValueSpan.Length;
+
+                    var propertyNameLength = valueLength <= scratchChars.Length ? reader.CopyString(scratchChars) : -1;
```

Comment:
talked with Todd offline and sounds like we can't do have a helper to or if/else here, so fine with this.